src/dialogs/more-info/controls/more-info-media_player.ts

```diff
@@ -231,12 +232,28 @@ class MoreInfoMediaPlayer extends LitElement {
   }
 
   private _handleClick(e: MouseEvent): void {
+    const action = (e.currentTarget! as HTMLElement).getAttribute("action")!;
     this.hass!.callService(
       "media_player",
-      (e.currentTarget! as HTMLElement).getAttribute("action")!,
-      {
-        entity_id: this.stateObj!.entity_id,
-      }
+      action,
+      action === "shuffle_set"
+        ? {
+            entity_id: this.stateObj!.entity_id,
+            shuffle: !this.stateObj!.attributes.shuffle,
+          }
+        : action === "repeat_set"
+        ? {
+            entity_id: this.stateObj!.entity_id,
+            repeat:
+              this.stateObj!.attributes.repeat === "all"
+                ? "one"
+                : this.stateObj!.attributes.repeat === "off"
+                ? "all"
+                : "off",
+          }
+        : {
+            entity_id: this.stateObj!.entity_id,
+          }
```

Comment:
This should be extracted into a helper.

Reply:
And we should use that helper in all places that use `computeMediaControls`

Reply:
So that I understand it right: Extract the service_calls into a exported method like computemediacontrols is, place that method inside media-player.ts and then pass that helper to the controlbuttons as a click callback when they are used?